cpp/CMakeLists.txt

```diff
@@ -244,51 +285,150 @@ add_library(cugraph SHARED
     ${CMAKE_CURRENT_BINARY_DIR}/gunrock/externals/moderngpu/src/mgpuutil.cpp
     ${CMAKE_CURRENT_BINARY_DIR}/gunrock/gunrock/util/gitsha1.c
 )
-
-option(BUILD_TESTS "Configure CMake to build tests" OFF)
+endif(NVGRAPH_LIGHT MATCHES True)
 
 ###################################################################################################
-# - link libraries --------------------------------------------------------------------------------
-
-target_link_libraries(cugraph cudart cuda cudf ${Boost_LIBRARIES} rmm)
+# - include paths ---------------------------------------------------------------------------------
 
-# Command to symlink files into the build directory
-add_custom_command(  # link the include directory
-    OUTPUT include
-    COMMAND ${CMAKE_COMMAND} -E create_symlink ${CMAKE_CURRENT_SOURCE_DIR}/include include
-    DEPENDS ${CMAKE_CURRENT_SOURCE_DIR}/include)
+target_include_directories(cugraph
+    PRIVATE
+    "${CMAKE_CUDA_TOOLKIT_INCLUDE_DIRECTORIES}"
+    "${Boost_INCLUDE_DIRS}"
+    "${RMM_INCLUDE}"
+    "${CUDF_INCLUDE}"
+    "${CMAKE_CURRENT_SOURCE_DIR}/../thirdparty"
+    "${CMAKE_CURRENT_SOURCE_DIR}/../thirdparty/cub"
+    "${CMAKE_CURRENT_SOURCE_DIR}/../thirdparty/cusp"
+    "${CMAKE_CURRENT_SOURCE_DIR}/src"
+    "${CMAKE_CURRENT_BINARY_DIR}/gunrock/"
+    "${CMAKE_CURRENT_BINARY_DIR}/gunrock/externals/moderngpu/include"
+    PUBLIC
+    "${CMAKE_CURRENT_SOURCE_DIR}/include"
+)
 
-# Configure the C++ tests
 ###################################################################################################
-# - add gtest -------------------------------------------------------------------------------------
-find_package(GTest QUIET)
-set_package_properties(GTest PROPERTIES TYPE OPTIONAL
-    PURPOSE "Google C++ Testing Framework (Google Test)."
-    URL "https://github.com/google/googletest")
-
+# - link libraries --------------------------------------------------------------------------------
 
+target_link_libraries(cugraph PRIVATE
+    ${CUDF_LIBRARY} ${RMM_LIBRARY} cublas cusparse curand cusolver cudart cuda ${Boost_LIBRARIES})
+if(OpenMP_CXX_FOUND)
+target_link_libraries(cugraph PRIVATE
 ###################################################################################################
-# - Generate the Tests -------------------------------------------------------------------------------
-if(GTEST_FOUND)
-    message(STATUS "Google C++ Testing Framework (Google Test) found in ${GTEST_ROOT}")
-    include_directories(${GTEST_INCLUDE_DIRS})
-    add_subdirectory(${CMAKE_CURRENT_SOURCE_DIR}/src/tests)
-else()
-    message(AUTHOR_WARNING "Google C++ Testing Framework (Google Test) not found: automated tests are disabled.")
-endif()
+###   Use ${OpenMP_CXX_LIB_NAMES} instead of OpenMP::OpenMP_CXX to avoid the following warnings.
+###
+###   Cannot generate a safe runtime search path for target TARGET_NAME
+###   because files in some directories may conflict with libraries in implicit
+###   directories:
+###   ...
+###
```

Comment:
That was a tricky one! Questions : 
1. In practice, which openmp is loaded for cuGraph now?
2. Are compiler and OpenMP versions interdependent? If not, is it safe to use a system compiler and a conda OpenMP and vice versa?
3. How safe is that for deployment? I mean are openMP headers and lib backward/forward compatible? If we build libcugraph.so using a very old/recent OpenMP, will the binary work for all openMP versions or should we add a note in the readme?

Reply:
1. So there are two answers: one for build time and another for install time. For the build time, assuming that LD_LIBRARY_PATH does not include a path to openmp, the conda one will be loaded (because the path to libcudf.so and librmm.so will go to BUILD_RPATH). For the install time, no RPATH(or RUNPATH) is specified, and libraries are searched in the following order 1) LD_LIBRARY_PATH 2) /etc/ld.so.conf 3) builtin directories - basically /lib and /usr/lib (https://gitlab.kitware.com/cmake/community/wikis/doc/cmake/RPATH-handling).

If you want to run with the system openmp in build time testing (without specifying LD_LIBRARY_PATH), we need to set BUILD_RPATH accordingly. I provided the instruction in the comment.

```
###   set(TARGET_BUILD_RPATH "")
###   foreach(TMP_VAR_FULLPATH IN LISTS OpenMP_CXX_LIBRARIES)
###       get_filename_component(TMP_VAR_DIR ${TMP_VAR_FULLPATH} DIRECTORY)
###       string(APPEND TARGET_BUILD_RPATH "${TMP_VAR_DIR};")
###       get_filename_component(TMP_VAR_REALPATH ${TMP_VAR_FULLPATH} REALPATH)
###       get_filename_component(TMP_VAR_DIR ${TMP_VAR_REALPATH} DIRECTORY)
###       # cmake automatically removes duplicates, so skip checking.
###       string(APPEND TARGET_BUILD_RPATH "${TMP_VAR_DIR};")
###   endforeach()
###   string(APPEND TARGET_BUILD_RPATH "${CONDA_PREFIX}/lib")
###   message(STATUS "TARGET_BUILD_RPATH=${TARGET_BUILD_RPATH}")
###   set_target_properties(target PROPERTIES
###       BUILD_RPATH "${TARGET_BUILD_RPATH}")
```

2. Newer compilers are necessary to use newer OpenMP features. If you use new OpenMP features (compiled using a newer compiler) and use an old library, it will not run (e.g. according to https://www.openmp.org/resources/openmp-compilers-tools/, "From GCC 6.1, OpenMP 4.5 is fully supported for C and C++."), but this affects only build time testing. On deployment, users should have a proper OpenMP library and that should appear first in the search order (than the outdated one if exists, see the answer for 1 for the search order).

3. This has no impact on deployment. No changes are made for installed one (cmake removes RUNPATH on installation unless we override the default option).

Reply:
Perfect. I checked the recent OpenMP releases note and I doubt we use any >4.0 features so we should be fine for any realistic configuration.